File: JavaScript/Baekjoon/1032.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const n = parseInt(input[0]);
const sLen = (input[1].trim()).length;

let ret = '';
for(let i = 0; i < sLen; i++) {
  let isCommon = true;
  for(let j = 2; j <= n; j++) {
    if(input[j-1][i] != input[j][i]) {
      isCommon = false;
      break;
    }
  }

  ret += isCommon ? input[1][i] : '?';
}

console.log(ret);

File: Python/Baekjoon/1032.py
n = int(input())
data = [input() for _ in range(n)]

result = ''
for i in range(len(data[0])):
    is_common = True
    for j in range(1, n):
        if data[j-1][i] != data[j][i]:
            is_common = False
            break
    if(is_common):
        result = result + data[0][i]
    else:
        result = result + '?'

print(result)